.github/labels.md

```diff
@@ -0,0 +1,234 @@
+label within the FreeOrion project.  When referencing to __Issues__ inside this
+documentation both Github Issues and Github Pull Requests are meant.
+
+
+## Purpose of labels
+
+Labels allow to attach common search and filter terms to Issues.  This makes it
+easier for developers, managers, reporters and to track the purpose, scope and
+state of Issues without explicit reading through possible length of comments,
+code reviews or commits.  Please follow this guide as close as possible to avoid
+duplicating work and wasting time for other developers and to enable better
+documentation on decisions made.
+
+
+## Label categories
+
+Labels are groupd in categories.  Those categories group together a common
+concept like the resolving state of an Issue or the target platform where the
+Issue occurs.  Categories share a prefix in the form `category_name:label_name`
+and are grouped together by color.  As a rule of thumb there is at least one
+label from every category that needs to be applied to an issue but use common
+sense to check if the label is approciate or not for the specific Issue.
+
+Currently there are three categories:
+
+
+#### Label category `category`
+
+Labels within this category describe is the intention of the Issue.  Usually
+there should be exact one intention for creating an Issue and it should not
+change during the lifetime of an Issue.
+
+#### Label category `component`
+
+The component lists the parts of the project this Issue affects or is affected
+by.  This should help developers to identify if they have required knowledge or
+interests to handle the issue.  There should be at least one component listed
+for an Issue.
+
+#### Label category `status`
+
+With the status category developers, managers or issue reporters can check if
+an issue needs further interaction and who should carry out this interaction.
+Issue have exactly one status.
+
+
+## Labels
+
+This section lists when an label should be applied and what actions may be taken
+additionally.
+
+
+#### Label `category:bug`
+
+The Issue describes or solves a percieved malfunction within the game.
+
+#### Label `category:feature`
+
+The Pull Request contains a new game feature.  Features are parts of the game
+which a player can interact with e.g.: A new species, a new gameplay mechanic
+or a new command line switch for the server that changes gameplay behaviour.
+
+#### Label `category:idea`
+
+??? No idea, overlaps with `category:feature` or `state:work in progress` IMO.
+
+#### Label `category:question`
+
+??? Questions don't belong into the issue tracker IMO.
+
+#### Label `category:refactoring`
+
+The Pull request contains a new or improved implementation of an existing
+feature or any other code restructuring that doesn't modify the current game
+mechanics or representation.
+
+#### Label `category:request`
+
+??? Questions don't belong into the issue tracker IMO.
+
+#### Label `category:tweak`
+
+??? No idea.
```

Comment:
I don't know if we need this then.  Developers tend to commit this kind of stuff onto master directly and Pull Requests for `code where it doesn't matter if it is applied or not` as in not features, refactoring or bug fixing are scarce IHMO.

And given your examples I would classify them as:

* refactoring in the case of code grooming.
* Feature or bug(fix) in case of the UI.  Even small changes are important enough to make it into the changelog so why not label them with significant labels?
* Same goes for your AI change example. Those issues are probably interesting for a player.